client/dashboard/emails/index.tsx

```diff
@@ -21,72 +20,60 @@ import './style.scss';
 			return { domainsWithEmails: [], domainsWithoutEmails: [] };
 		}
 
-		// We filter the same way v1 does.
-		const nonWpcomDomains = domains.filter(
-			( domain ) => domain.subtype.id !== DomainSubtype.DEFAULT_ADDRESS
-		);
+		const domainsWithEmails =
+			allEmailAccounts?.map( ( mailbox ) => mailbox.domains[ 0 ].domain ) ?? [];
+		return {
+			domainsWithEmails: domains.filter( ( d ) => domainsWithEmails.includes( d.domain ) ),
+			domainsWithoutEmails: domains.filter( ( d ) => ! domainsWithEmails.includes( d.domain ) ),
+		};
+	}, [ allEmailAccounts, domains, isLoadingDomains, isLoadingEmailAccounts ] );
 
-		const domainsWithEmails = nonWpcomDomains.filter( domainHasEmail ) as Domain[];
-		const domainsWithoutEmails = nonWpcomDomains.filter(
-			( domain ) => ! domainHasEmail( domain )
-		) as Domain[];
-
-		return { domainsWithEmails, domainsWithoutEmails };
-	}, [ domains, isLoadingDomains ] );
-
-	const mailboxQueries = useQueries( {
-		queries: domainsWithEmails.map( ( domain: Domain ) =>
-			mailboxAccountsQuery( domain.blog_id, domain.domain )
-		),
-	} );
-
-	const isLoadingMailboxes = mailboxQueries.some( ( q ) => q.isLoading );
-
-	// Build emails by pairing each domain with its corresponding mailboxes query
 	const emails: Email[] = useMemo( () => {
-		if ( ! domainsWithEmails.length ) {
+		if ( ! allEmailAccounts?.length ) {
 			return [];
 		}
-		return domainsWithEmails.flatMap( ( domain, index ) => {
-			const emailAccounts = ( mailboxQueries[ index ]?.data as EmailAccount[] ) ?? [];
-
-			const skipFilterForwards = emailAccounts.length === 1;
-
-			return emailAccounts
-				.filter( ( account ) => skipFilterForwards || account.account_type !== 'email_forwarding' )
-				.flatMap( ( account ) =>
-					account.emails.map( ( box: EmailBox ) => mapMailboxToEmail( box, account, domain ) )
-				)
-				.filter( ( email ) => email.canUserManage ) as Email[];
-		} );
-	}, [ domainsWithEmails, mailboxQueries ] );
+		return allEmailAccounts
+			.flatMap( ( account ) =>
+				account.emails.map( ( box: EmailBox ) => mapMailboxToEmail( box, account ) )
+			)
+			.filter( ( email ) => email.canUserManage ) as Email[];
+	}, [ allEmailAccounts ] );
 
 	// Gather domains with unused mailbox warnings
 	const domainsWithUnusedMailbox: string[] = useMemo( () => {
-		if ( ! domainsWithEmails.length ) {
+		if ( ! allEmailAccounts?.length ) {
 			return [];
 		}
-		return domainsWithEmails.reduce< string[] >( ( acc, domain, index ) => {
-			const emailAccounts = ( mailboxQueries[ index ]?.data as EmailAccount[] ) ?? [];
-			const hasUnusedWarning = emailAccounts.some( ( account ) =>
-				( account.warnings ?? [] ).some(
-					( w ) => w.warning_slug === 'unused_mailboxes' && w.warning_type === 'notice'
-				)
+		const warnedDomains = new Set< string >();
+		for ( const account of allEmailAccounts ) {
+			const hasUnusedWarning = ( account.warnings ?? [] ).some(
+				( w ) => w.warning_slug === 'unused_mailboxes'
```

Comment:
Given that we are not checking to the warning type anymore, should the check on the router be updated as well?

https://github.com/Automattic/wp-calypso/blob/0f94bf1c7e1084e375c22fd9e4f2ca3a8b54025c/client/dashboard/app/router/emails.tsx#L206-L233